internal/testing/setup/setup.go

```diff
@@ -181,23 +182,21 @@ func NewAzureTestPipeline(ctx context.Context, t *testing.T, accountName string,
 	}
 
 	httpClient = azureHTTPClient(r)
-	p := newPipeline(credential, r)
+	p := newPipeline(credential, api, r)
 
 	return p, done, httpClient
 }
 
-func newPipeline(c azblob.Credential, r *recorder.Recorder) pipeline.Pipeline {
+func newPipeline(c azblob.Credential, api string, r *recorder.Recorder) pipeline.Pipeline {
 	if c == nil {
 		panic("pipeline credential can't be nil")
 	}
 
 	f := []pipeline.Factory{
 		// sets User-Agent for recorder
 		azblob.NewTelemetryPolicyFactory(azblob.TelemetryOptions{
-			Value: "X-Az-Target",
```

Comment:
I'm not sure what the "X-Az-Target" here meant; it was getting prepended to the User-Agent header.

Reply:
Yeah, unintended. And to your other point, its is not needed for the tests and can be removed all together.